vizro-ai/tests/unit/vizro-ai/dashboard/_response_models/test_components.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+from vizro_ai.dashboard._response_models.components import ComponentPlan
+
+
+class TestComponentCreate:
+    """Tests component creation."""
+
+    def test_component_plan_instantiation(self):
+        component = ComponentPlan(
+            component_id="card_1",
+            component_type="Card",
+            component_description="This is a card",
+            df_name="N/A",
+        )
+        assert component.component_id == "card_1"
+        assert component.component_type == "Card"
+        assert component.component_description == "This is a card"
+        assert component.df_name == "N/A"
+
+    @pytest.mark.xfail(raises=ValueError, reason="Known issue: real model is required for .plot")
```

Comment:
Could we not mock the response here? I'd be very interested in getting this type of test to work. Essentially fake the return of the real call to what the card should be. Like that we can also test what the failure modes etc are.

Reply:
Unfortunately we cannot fake the call, as real model is required for `.plot()`. 

line 48 in `components.py` -> `vizro_ai = VizroAI(model=model)` cannot be circumvented. 

If I try to pass fake llm model, I will get Vizro-Ai error:
` f"Model {model} not found! List of available model can be found at ...`

This is the reason why we cannot tests some other parts of the code as well. Likewise, the failure modes cannot be tested, as we will never reach them, and instead get "Model not found" error.

Reply:
Could we not Mock the call to VizroAI.plot() not the model? I mean we are really not interested in testing the plotting function right?

Reply:
I have provided an example in my latest commit, does that make sense to you?